xbmc/Application.cpp

```diff
@@ -2486,7 +2486,7 @@ void CApplication::FrameMove(bool processEvents, bool processGUI)
 #if defined(TARGET_RASPBERRY_PI) || defined(HAS_IMXVPU)
     // This code reduces rendering fps of the GUI layer when playing videos in fullscreen mode
     // it makes only sense on architectures with multiple layers
-    if (g_graphicsContext.IsFullScreenVideo() && !m_pPlayer->IsPausedPlayback())
+    if (g_renderManager.IsVideoLayer() && g_graphicsContext.IsFullScreenVideo() && !m_pPlayer->IsPausedPlayback())
```

Comment:
the call to IsFullScreenVideo is less expensive and should be first, IsVideoLayer last